### PR TITLE
Add custom configuration to jerry_parse and its variants

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

File: tests/unit-core/test-api.c [hunk 1]
```diff
@@ -803,12 +839,10 @@ main (void)
   jerry_release_value (val_t);
 
   /* Test: create function */
-  const jerry_char_t func_resource[] = "unknown";
   const jerry_char_t func_arg_list[] = "a , b,c";
   const jerry_char_t func_src[] = "  return 5 +  a+\nb+c";
 
-  jerry_value_t func_val = jerry_parse_function (func_resource,
-                                                 sizeof (func_resource) - 1,
+  jerry_value_t func_val = jerry_parse_function (NULL,
```
Comment (zherczeg) [inline tests/unit-core/test-api.c:845]: This resource has no function, so I decided to remove it.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -271,7 +274,6 @@ Flags for [jerry_generate_snapshot](#jerry_generate_snapshot) and
 [jerry_generate_function_snapshot](#jerry_generate_function_snapshot) functions:
 
  - JERRY_SNAPSHOT_SAVE_STATIC - generate static snapshot (see below)
- - JERRY_SNAPSHOT_SAVE_STRICT - strict source code provided
```
Comment (zherczeg) [inline docs/02.API-REFERENCE.md:274]: Strict option is passed as part of parse options.

File: jerry-core/ecma/base/ecma-globals.h [hunk 3]
```diff
@@ -331,6 +331,32 @@ typedef struct ecma_native_pointer_t
   struct ecma_native_pointer_t *next_p; /**< points to the next ecma_native_pointer_t element */
 } ecma_native_pointer_t;
 
+/**
+ * Option bits for ecma_parse_options_t.
+ */
+typedef enum
+{
+  /* ECMA_PARSE_STRICT_MODE = (1 << 0) */
+  /* ECMA_PARSE_MODULE = (1 << 1) */
```
Comment (rerobika) [inline jerry-core/ecma/base/ecma-globals.h:340]: This commented code looks kinda weird for me. Is this intentional?

Comment (zherczeg) [reply jerry-core/ecma/base/ecma-globals.h:340]: I wanted to emphasize that those constants are defined elsewhere. Do you have a better suggestion?

Comment (rerobika) [reply jerry-core/ecma/base/ecma-globals.h:340]: How about just a simple comment like:

The first 2 bits are restricted for ...., see ...

Comment (zherczeg) [reply jerry-core/ecma/base/ecma-globals.h:340]: What about:   `/* bit 0: ECMA_PARSE_STRICT_MODE */` ?

**Comment by zherczeg** [comment]
Thank you for the review, patch is updated.

**Review by rerobika — APPROVED** [review]

**Review by galpeter — APPROVED** [review]
lgtm

**Pull request merged**